lib/Partitioner/Partitioner.cpp

```diff
@@ -492,8 +492,7 @@ void Partitioner::doPartitioning(Function *F, NodeToFunctionMap &mapping) {
 
   // Push the DAG {root, nodes} into partitions_.
   //  DAG newDag = {.root = std::move(DAGRoot), .nodes = std::move(nodes)};
-  partitions_.push_back(
-      {.root = std::move(DAGRoot), .nodes = std::move(nodes)});
+  partitions_.push_back({std::move(DAGRoot), std::move(nodes)});
```

Comment:
This seems a bit fragile to depend on the default constructor. Maybe create a new `DAG` and initialize each member or create a constructor for DAG?

Reply:
@jackm321 Maybe just use emplace_back then?

Reply:
@ayermolo emplace_back using the same constructor would be the same issue. If you create a `DAG dag` and do emplace_back(std::move(dag)) that would be fine

Reply:
Changed to emplace_back so only the explicit argument constructor is used. Added the move constructor for entire DAG in case push_back, or something similar, is used elsewhere.